routes/auth/auth.routes.js

```diff
@@ -0,0 +1,141 @@
+const User = require('../../models/User.model');
+
+const router = require('express').Router();
+const mongoose = require('mongoose');
+
+const passport = require('passport');
+const LocalStrategy = require('passport-local');
+
+const bcryptjs = require('bcryptjs');
+const saltRounds = 10;
+
+// require auth middleware
+const { isLoggedIn, isLoggedOut } = require('../../middleware/middleware.js');
+
+// https://github.com/howardmann/authentication
+passport.use(
+  'local-login',
+  new LocalStrategy(
+    {
+      // Fields to accept
+      usernameField: 'username',
+      passwordField: 'password',
+      passReqToCallback: true, // allows us to access req in the call back
+    },
+    async (req, username, password, done) => {
+      username = req.sanitize(username);
+      password = req.sanitize(password);
+      // Check if user and password is valid
+      let user = await User.findOne({ username });
+      let passwordValid = user && bcryptjs.compareSync(password, user.passwordHash);
+      // If password valid call done and serialize user.id to req.user property
+      if (passwordValid) {
+        return done(null, {
+          id: user.id,
+          name: user.username,
+        });
+      }
+      return done(null, false);
+    }
+  )
+);
+
+passport.serializeUser(function (user, cb) {
+  process.nextTick(function () {
+    cb(null, { id: user.id, username: user.username });
+  });
+});
+
+passport.deserializeUser(function (user, cb) {
+  process.nextTick(function () {
+    return cb(null, user);
+  });
+});
+
+// Login Routes
+router.get('/login', isLoggedOut, function (req, res, next) {
+  try {
+    res.render('auth/login', { message: req.flash('error') });
+  } catch (err) {
+    console.error('Sorry, there was an error: ', err);
+    res.render('error');
+  }
+});
+
+router.post(
+  '/login',
+  passport.authenticate('local-login', {
+    successReturnToOrRedirect: '/dashboard',
+    failureRedirect: '/login',
+    successFlash: true,
+    failureFlash: true,
+    successFlash: 'Succesfull!',
+    failureFlash: 'Invalid username or password.',
+  })
+);
+
+// Signup Routes
+router.get('/signup', isLoggedOut, (req, res, next) => {
+  try {
+    res.render('auth/signup');
+  } catch (err) {
+    console.error('Sorry, there was an error: ', err);
+    res.render('error');
+  }
+});
+
+router.post('/signup', async (req, res, next) => {
+  const username = req.sanitize(req.body.username);
+  const password = req.sanitize(req.body.password);
+  if (!username || !password) {
+    res.render('auth/signup', { errorMessage: 'All fields are mandatory. Please provide your username, email and password.' });
+    return;
+  }
+
+  const regex = /(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,}/;
```

Comment:
give this regex a nice name that describes what it's doing.

Reply:
from the way it's used I would guess it's some form of password validator, but having a good name here, would mean I didn't have to guess ^^